cishouseholds/derive.py

```diff
@@ -1497,6 +1497,64 @@ def assign_date_difference(
         raise TypeError(f"{format} format not supported")
 
 
+def derive_digital_merge_type(
+    df: DataFrame,
+    column_name_to_assign: str,
+) -> DataFrame:
+    """
+    Calculate the difference in days between two dates.
+    From households_aggregate_processes.xlsx, derivation number 27.
+
+    Parameters
+    ----------
+    df
+    survey_completion_status
+        Variable with survey completion status.
+    start_reference_column
+        First date column name.
+    end_reference_column
+        Second date column name.
+
+    Return
+    ------
+    pyspark.sql.DataFrame
+    """
+    df = df.withColumn(
+        column_name_to_assign,
+        F.when(
+            (F.col("form_start_datetime").isNull())
+            & (F.col("participant_completion_window_end_datetime").isNull())
+            & (F.col("Survey_completion_status").isin(["Submitted", "Completed"])),
+            "Matched",
+        )
+        .when(
+            (~F.col("form_start_datetime").isNull())
+            & (F.to_date(F.col("participant_completion_window_end_datetime")) > F.to_date(F.col("file_date")))
+            & (F.col("survey_completion_status").isin(["Partially Completed", "New"])),
+            "Temporary Orphan",
+        )
+        .when(
+            (F.col("form_start_datetime").isNull())
+            & (F.to_date(F.col("participant_completion_window_end_datetime")) > F.to_date(F.col("file_date")))
+            & (F.col("survey_completion_status") == "New"),
+            "Potential Orphan",
+        )
+        .when(
+            (~F.col("form_start_datetime").isNull())
```

Comment:
remove reference to form_start_datetime for this match type (row 2 in specification table). Requirement does not rely on this field. 

Also need the third 'Matched' type scenario adding. Where form_start_datetime is not null, completion_window <=file_date and completion_status = new